### PR TITLE
♻️ peformance-impl: simplify tick()

### PR DESCRIPTION
**Pull request by samouri**
- places all of `tick()`'s branched logic in a single place near the top, instead of being spread throughout.
- only sends `value` or `delta`, instead of both.

### CODE REVIEW

File: src/service/performance-impl.js [hunk 1]
```diff
@@ -715,30 +715,27 @@ export class Performance {
...
-      'delta': opt_delta != null ? Math.max(opt_delta, 0) : undefined,
-    });
     if (this.isMessagingReady_ && this.isPerformanceTrackingOn_) {
       this.viewer_.sendMessage('tick', data);
     } else {
       this.queueTick_(data);
     }
-    // Mark the event on the browser timeline, but only if there was
-    // no delta (in which case it would not make sense).
-    if (arguments.length == 1) {
```
Comment (samouri) [inline src/service/performance-impl.js]: Why were we checking `arguments.length == 1` as opposed to for a null/undefined opt_delta?

File: src/service/performance-impl.js [hunk 2]
```diff
@@ -715,30 +715,27 @@ export class Performance {
...
-      this.mark(label);
-    }
 
-    // Store certain page visibility metrics to be exposed as analytics
-    // variables.
-    const storedVal = Math.round(
```
Comment (samouri) [inline src/service/performance-impl.js]: Why were we rounding when the value could not be a decimal anyway? (`Date.now()` never returns numbers with fractional components)

File: src/service/performance-impl.js [hunk 3]
```diff
@@ -715,30 +715,27 @@ export class Performance {
    *     this directly.
    */
   tick(label, opt_delta) {
-    const value = opt_delta == undefined ? this.win.Date.now() : undefined;
+    let data;
+    let storedVal;
+
+    // Absolute value case (not delta).
+    if (typeof opt_delta === 'undefined' || opt_delta === null) {
```
Comment (jridgewell) [inline src/service/performance-impl.js]: `typeof` isn't necessary for local variables, it'll never throw an error. We can just do `opt_delta == null`

Comment (samouri) [reply src/service/performance-impl.js]: do you mean `opt_delta == undefined`?

File: src/service/performance-impl.js [hunk 4]
```diff
@@ -715,30 +715,27 @@ export class Performance {
...
+
+    // Absolute value case (not delta).
+    if (opt_delta == undefined) {
+      // Marking only makes sense for non-deltas.
+      this.mark(label);
+      const now = this.win.Date.now();
+      data = dict({'label': label, 'value': now});
+      storedVal = now - this.initTime_;
+    } else {
+      data = dict({'label': label, 'delta': Math.max(opt_delta, 0)});
```
Comment (choumx) [inline src/service/performance-impl.js]: Does this mean we no longer send `data.value` to `sendMessage()` and `queueTick_()`?

File: src/service/performance-impl.js [hunk 5]
```diff
@@ -715,30 +715,27 @@ export class Performance {
...
-    // no delta (in which case it would not make sense).
-    if (arguments.length == 1) {
-      this.mark(label);
-    }
 
-    // Store certain page visibility metrics to be exposed as analytics
-    // variables.
-    const storedVal = Math.round(
-      opt_delta != null ? Math.max(opt_delta, 0) : value - this.initTime_
-    );
```
Comment (choumx) [inline src/service/performance-impl.js]: I actually find this less readable. I think there's a ToTT for it somewhere but I also found this: https://refactoring.com/catalog/reduceScopeOfVariable.html

File: src/service/performance-impl.js (hunk 4 above)
Comment (samouri) [reply src/service/performance-impl.js]: Correct. Isn't that better?

File: src/service/performance-impl.js (hunk 5 above)
Comment (samouri) [reply src/service/performance-impl.js]: You prefer the `opt_delta` condition repeated 3 times?

Comment (samouri) [reply src/service/performance-impl.js]: Speaking of this line, I have a question. Why in the absolute case (relative to epoch) do we set `storedVal` to the delta?

File: src/service/performance-impl.js (hunk 4 above)
Comment (choumx) [reply src/service/performance-impl.js]: Oh I see, lol. Thanks for cleaning this up.

Minor suggestion:
```js
const data = dict({'label': label});
if (...) {
  data['value'] = now;
} else {
  data['delta'] = Math.max(opt_delta, 0);
}
```

File: src/service/performance-impl.js (hunk 5 above)
Comment (samouri) [reply src/service/performance-impl.js]: answer: its used in the url-rewriter, and non-viewer use cases care more about relative than absolute times.

**Pull request merged**